core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -591,6 +597,9 @@ class DiplomacyManager() : IsPartOfGameInfoSerialization {
                         if (!otherCivDiplomacy().hasFlag(DiplomacyFlags.ResearchAgreement))
                             scienceFromResearchAgreement()
                     }
+                    DiplomacyFlags.DefensivePact.name -> {
```

Comment:
What is this for?

Reply:
The trade creates the flag, which counts down to zero and then cancels the defensive pact by setting the DiplomaticStatus back to peace. Previously the diplomaticStatus was set to DefensivePact.

Reply:
Gotcha